core/src/BibleEngine.class.ts

```diff
@@ -885,6 +887,11 @@ export class BibleEngine {
                 )
                     throw new Error(`can't add phrases: normalisation failed`);
 
+                if (!globalState.isWithinParagraph)
+                    console.error(
+                        `can't add phrase "${content.content}" (${book.osisId} ${content.versionChapterNum}:${content.versionVerseNum}): not within a paragraph`
```

Comment:
At the osis level, CUV doesn't have paragraphs to wrap phrases. Is this a problem? I suppose the osis parser can add a paragraph wrapper, if none is found when parsing a phrase

Reply:
good question. maybe we add a the property `hasParagraphs` to the options of the importer. if `false` we don't fail here?

Reply:
Hmm... Its tricky. Maybe best to remove this error?

For instance, what if a phrase is in a section heading, or so on?

Don't think every phrase should need to be within a paragraph

(we could add the flag, definitely, just want to make sure it truly preserves data integrity by throwing this error)

Reply:
there can't be a phrase in a section heading. a phrase is by definition canonical bible text.

thing is, if a version uses paragraphs, a phrase that is without a paragraph points to an error in the file. having strict error checking like this pointed me to a lot of issues in the source files (or sometimes the parser) automatically. those are things you can't find otherwise (until some users report an issue with the text rendering).

Reply:
i think its good to have a high consistency in the document structure of bible engine. enforcing this early on makes things simpler and more stable further down

Reply:
i added an `isPlaintext` attribute to `IBibleVersionEntity`. this might come in handy for other optimisations for that kind of versions.